src/dodal/common/udc_directory_provider.py

```diff
@@ -7,11 +8,14 @@
 
 
 class PandASubdirectoryProvider(UpdatingDirectoryProvider):
+    """Directory provider for the HDFPanda. Points to, and optionally creates, a panda subdirectory relative to the
+    directory path provided"""
+
     resource_dir = Path("panda")
 
     def __init__(self, directory: Path | None = None):
         if directory is None:
-            LOGGER.warn(
+            LOGGER.debug(
```

Comment:
I changed this to a debug since i03 initialises with directory=None, so those messages would get very annoying